crates/bin/src/args.rs

```diff
@@ -223,10 +223,28 @@ pub struct Args {
     #[clap(help_heading = "Options", long, alias = "roots")]
     pub root: Option<PathBuf>,
 
-    /// The URL of the registry index to use
+    /// The URL of the registry index to use.
+    ///
+    /// Cannot be used with `--registry`.
```

Comment:
Clap has support for defining exclusions, we could use that on this pair (and other pairs) of mutually exclusive options.

Reply:
(In follow up PR)

Reply:
(In follow up PR)